### PR TITLE
google-cloud page: add links to HTTP server docs and samples

### PR DESCRIPTION
**Pull request by kevmoo**
Drop link to dartbug – the code is not well documented
Clarify the App Engine bits

### CODE REVIEW

**Comment by kevmoo** [comment]
CC @mit-mit

File: src/server/google-cloud.md [hunk 1]
```diff
@@ -19,9 +22,8 @@ You can use Cloud Run's flexible container support,
 combined with Dart's Docker images,
 to run server-side Dart code.
```
Comment (kwalrath) [inline src/server/google-cloud.md:24]:
```suggestion
```

File: src/server/google-cloud.md [hunk 2]
```diff
@@ -9,6 +9,9 @@ Dart servers can use many
 often with the help of pre-packaged [Docker images][]
 that the Dart team maintains.
```
Comment (kwalrath) [inline src/server/google-cloud.md:11]:
```suggestion
```

**Review by domesticmouse — APPROVED** [review]

File: src/server/google-cloud.md [hunk 3]
```diff
@@ -19,9 +22,8 @@ You can use Cloud Run's flexible container support,
 combined with Dart's Docker images,
 to run server-side Dart code.
 
-One example of a Dart server that uses Cloud Run is dartbug.com,
-the redirect service for Dart issues and bugs.
-You can find its source code in the [dart-lang/dartbug.com repo][].
+For examples of Dart servers implemented to run on Cloud Run, see the
+[dart-lang/samples repo][].
```
Comment (kwalrath) [inline src/server/google-cloud.md:26]:
```suggestion
Examples of Dart servers implemented to run on Cloud Run are
[in the dart-lang/samples/repo][server examples].
```

File: src/server/google-cloud.md [hunk 4]
```diff
@@ -45,16 +47,16 @@ For more information, see the [GKE overview][].
...
+server-side Dart code.
+If you _want_ to use App Engine, consider using the [`appengine` package][].
 
 
+[App Engine]: https://cloud.google.com/appengine
 [`appengine` package]: {{site.pub-pkg}}/appengine
 [ce]: https://cloud.google.com/compute/docs/containers
 [cr]: https://cloud.google.com/run/docs/quickstarts/build-and-deploy/other
-[dart-lang/dartbug.com repo]: https://github.com/dart-lang/dartbug.com
+[dart-lang/samples repo]: https://github.com/dart-lang/samples/tree/master/server
```
Comment (kwalrath) [inline src/server/google-cloud.md:60]:
```suggestion
[server examples]: https://github.com/dart-lang/samples/tree/master/server
```

**Review by kwalrath — APPROVED** [review]
Thanks for the PR! I'd like some tweaks to tighten things up.

**Pull request merged**